src/widgets/ot_color_picker.eliom

```diff
@@ -0,0 +1,183 @@
+[%%shared
+
+open Eliom_content.Html
+open Eliom_content.Html.F
+
+type div = Html_types.div Eliom_content.Html.D.elt
+type t = (string ref * div * div list * div)
+
+let raise_color_samples_exception () =
+  let message = "Ot_color_picker.generate_color_samples, \
+                 the argument have to be greater than 1" in
+  invalid_arg message
+
+let generate_color_samples precision =
+
+  let color_list = match precision with
+    | p when p <= 1     -> raise_color_samples_exception ()
+    | precision         ->
+      let step = 255 / (precision - 1) in
+      let rec aux_build nl v =
+        if (v > 255)
+        then nl
+        else aux_build ((Printf.sprintf "%02X" v)::nl) (v + step)
+      in aux_build [] 0
+  in
+  let max_iteration = List.length color_list in
+
+  let rec aux_red nl = function
+    | n when n >= max_iteration -> nl
+    | n                         ->
+      let red = List.nth color_list n in
+
+      let rec aux_green nl = function
+        | n when n >= max_iteration     -> nl
+        | n                             ->
+          let green = List.nth color_list n in
+
+          let rec aux_blue nl = function
+            | n when n >= max_iteration -> nl
+            | n                         ->
+              let blue = List.nth color_list n in
+              aux_blue (("#" ^ red ^ green ^ blue)::nl) (n + 1)
+
+          in aux_green ((aux_blue [] 0)::nl) (n + 1)
+
+      in aux_red ((aux_green [] 0)::nl) (n + 1)
+
+  in aux_red [] 0
+
+
+(* Some pre-generated color samples in several precisions. *)
+let color_samples_p2 = lazy (generate_color_samples 2)
+let color_samples_p3 = lazy (generate_color_samples 3)
+let color_samples_p4 = lazy (generate_color_samples 4)
+let color_samples_p5 = lazy (generate_color_samples 5)
+let color_samples_p6 = lazy (generate_color_samples 6)
+
+(* Some hand-mained color samples *)
+let color_samples_10 = [[["#E03625"; "#FF4B3A"];
+                         ["#FF7E02"; "#FFC503"];
+                         ["#01CD64"; "#AF58B9"];
+                         ["#0198DD"; "#254760"];
+                         ["#FFFFFF"; "#000000"]]]
+
+let color_samples_6 = [[["#BEC3C7"; "#7A8E8D"];
+                        ["#1C3D50"; "#0280B4"];
+                        ["#00A385"; "#A444B2"]]]
+
+
+(* Take one list of list of list of color (string) and build table list with it.
+   return also div_color_list to allow to launch start script detection. *)
+let generate_color_table color_samples =
```

Comment:
Takes a list of lists of lists of colors (strings) and returns a table list. Also returns a `div_color_list` for launching start script detection.

The singular/plural needs to be fixed in the `.eliomi`.